app/uk/gov/hmrc/apisubscriptionfields/repository/SubscriptionFieldsRepository.scala

```diff
@@ -74,18 +84,23 @@ class SubscriptionFieldsMongoRepository @Inject()(mongoDbProvider: MongoDbProvid
     )
   }
 
-  //TODO change return type to boolean
-  override def save(subscription: SubscriptionFields): Future[Unit] = {
-    val selector = selectorById(subscription.id)
-    Logger.debug(s"[save] selector: $selector")
-    collection.find(selector).one[BSONDocument].flatMap {
-      case Some(document) => collection.update(selector = BSONDocument("_id" -> document.get("_id")), update = subscription)
-      case _ => collection.insert(subscription)
-    }.map {
-      writeResult => handleError(writeResult, s"Could not save subscription fields: $subscription")
+  /**
+    * Saves or inserts entity depending on if it already exists.
+    * Returns Future of isInserted Boolean flag if everything went OK - otherwise a failed Future with an error message.
+    * @param subscription entity to upsert
+    */
+  def upsert(subscription: SubscriptionFields): Future[Boolean] = {
+    collection.update(selector = BSONDocument("_id" -> subscription.id), update = subscription, upsert = true).map {
```

Comment:
Why did you change the code here?

Also, are you sure this code is correct?
`_id` is the mongo primary key.
`id` is the field we defined.
They are two different things.

Reply:
done